amen/audio.py

```diff
@@ -21,8 +22,35 @@ def __init__(self, file_path, convert_to_mono=False, sample_rate=22050):
         self.raw_samples = y
         self.num_channels = y.ndim
         self.duration = librosa.get_duration(y=y, sr=sr)
+        self.timings = self.create_timings()
         self.features = self.create_features()
 
+    def create_timings(self):
+        """
+        Create timings in a timings dict.
+        """
+        timings = {}
+        timings['beats'] = TimingList('beats', self.get_beats(), self)
+        return timings
+
+    def get_beats(self):
+        """
+        Gets beats using librosa's beat tracker.
+        """
+        y_mono = librosa.to_mono(self.raw_samples)
+        tempo, beat_frames = librosa.beat.beat_track(
+            y=y_mono, sr=self.sample_rate, trim=False)
+        # convert frames to times
+        beat_times = librosa.frames_to_time(beat_frames, sr=self.sample_rate)
+        # make the list of (start, duration)s that TimingList expects
+        starts_durs = []
+        for i, start in enumerate(beat_times[:-1]):
+            starts_durs.append((start, beat_times[i+1] - start))
+        # now get the last one
+        starts_durs.append((beat_times[-1], self.duration - beat_times[-1]))
+
```

Comment:
you can do this bit a bit more succinctly as follows:

```
# pad beat_times to full duration
beat_times = librosa.util.fix_frames(beat_times, x_min=0, x_max=self.duration)
# Generate all inter-beat intervals
starts_durs = [(s, t-s) for (s,t) in zip(beat_times, beat_times[1:])]
```

The above code will pad out to span the range `[0, duration]` and then slice adjacent indices to make intervals.  The benefit of doing it this way is that it handles things like duplicated event times at the boundaries.  If you don't want to add the [0, beat_1] interval, just set `x_min=None`.